Modules/Core/SpatialObjects/include/itkTubeSpatialObject.hxx

```diff
@@ -345,7 +345,7 @@ TubeSpatialObject< TDimension, TTubePointType >
     l = std::sqrt(l);
     // if the adjacent points correspond, use the current point and one
     //   forward point
-    if ( Math::AlmostEquals( l, 0.0 ) )
+    if ( Math::AlmostEquals( l, 0.0 ) || l != l )
```

Comment:
`||    l != l`

This seems like an always false statement that will cause compiler warnings about autological comparisons.

Was this used for debugging?

Reply:
Using `std::isnan` is far clearer, if that is what you wanted to check.

Reply:
Changed to std::isnan() call.